charms/openfga-k8s/src/charm.py

```diff
@@ -264,10 +281,11 @@ def _update_workload(self, event):
         env_vars["OPENFGA_DATASTORE_ENGINE"] = "postgres"
         env_vars["OPENFGA_DATASTORE_URI"] = self._state.db_uri
 
-        secret = self.model.get_secret(id=self._state.token_secret_id)
-        secret_content = secret.get_content()
+        token = self._get_token(event)
+        if not token:
+            return
```

Comment:
Returning here without setting the status may cause a misleading status on the JIMM unit.
We should either,
- Don't return here and allow the check lower down that would set the unit status to `Blocked` to take effect.
- Set the unit status here and log appropriately.